.github/workflows/integration-local.yml

```diff
@@ -46,55 +46,122 @@ jobs:
-          env:
-            CYPRESS_RECORD_KEY: ${{secrets.CYPRESS_RECORD_KEY}}
-            GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
-
-        - uses: actions/upload-artifact@v3
-          if: failure()
-          with:
-            name: cypress-screenshots
-            path: cypress/screenshots
-        # Test run video was always captured, so this action uses "always()" condition
-        - uses: actions/upload-artifact@v3
-          if: always()
-          with:
-            name: cypress-videos
-            path: cypress/videos
\ No newline at end of file
+    name: Kitty Items E2E test on emulator
+    timeout-minutes: 10
+    runs-on: ubuntu-latest
+    container: cypress/browsers:node16.14.2-slim-chrome103-ff102 #use arm64 arch on Apple M1 chips when running locally
+    steps:
+      - name: Install build dependencies (mostly for sqlite on ARM)
+        run: apt-get update -y && apt-get install build-essential -y 
+
+      - name: Install Flow CLI
+        run: | 
+          sh -ci "$(curl -fsSL https://storage.googleapis.com/flow-cli/install.sh)" &&
+          echo "/root/.local/bin:/github/home/.local/bin" >> $GITHUB_PATH
+          echo $HOME &&
+          ls -l $HOME/.local/bin &&
+          chmod a+r $HOME/.local/bin/flow &&
+          ls -l $HOME/.local/bin &&
+          $HOME/.local/bin/flow version
+
+      - name: Checkout
+        uses: actions/checkout@v3
+      
+      - name: chmod subdirs to bypass lerna exec access permission issues
+        run: |
+          pwd &&
+          ls -l && 
+          chmod -R 777 api &&
+          chmod -R 777 web &&
+          chmod -R 777 cadence/tests &&
+          ls -l 
+
+      - name: Cypress run emulator
+        uses: cypress-io/github-action@v4
+        with:
+          start: npm run dev:emulator
+          wait-on: http://localhost:3001
+          spec: cypress/e2e/emulator.cy.js
+          record: true
+          browser: chrome
+        env:
+          CYPRESS_RECORD_KEY: ${{secrets.CYPRESS_RECORD_KEY}}
+          GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
+
+      - uses: actions/upload-artifact@v3
+        if: failure()
+        with:
+          name: cypress-screenshots
+          path: cypress/screenshots
+      # Test run video was always captured, so this action uses "always()" condition
+      - uses: actions/upload-artifact@v3
+        if: always()
+        with:
+          name: cypress-videos
+          path: cypress/videos
+
+  cypress-e2e-testnet:
+    name: Kitty Items E2E test on Testnet
+    needs: cypress-e2e-emulator # we don't want both jobs to run in parallel
+    timeout-minutes: 10
+    runs-on: ubuntu-latest
+    container: cypress/browsers:node16.14.2-slim-chrome103-ff102 #use arm64 arch on Apple M1 chips when running locally
+    env:
+      ADMIN_ADDRESS: ${{secrets.ADMIN_ADDRESS_E2E}}
+      FLOW_PRIVATE_KEY: ${{secrets.FLOW_PRIVATE_KEY_E2E}}
+      FLOW_PUBLIC_KEY: ${{secrets.FLOW_PUBLIC_KEY_E2E}}
+      E2E_TESTING: ${{secrets.E2E_TESTING}}
```

Comment:
instead of using E2E_TESTING, use existing github actions context info

Reply:
Technically, this just needs to be an indicator/flag, so we don't really need any contextual info. I set this as the name of the job, and I simply check that this flag was set.